Winium/Winium.StoreApps.InnerServer/Element/WiniumElement.IsUserVisible.cs

```diff
@@ -82,6 +89,42 @@ internal bool IsUserVisible()
             }
         }
 
+        private CommandBar FindCommandBar(AppBarButton element)
+        {
+            DependencyObject parent = element;
+
+            while ( parent != null  )
+            {
+                parent = VisualTreeHelper.GetParent(parent);
+
+                FrameworkElement parentElement = parent as FrameworkElement;
+
+                if (parentElement != null && parentElement.Visibility == Visibility.Collapsed )
+                    return null;
+
+                if (parent.GetType() == typeof(AppBar) || parent.GetType() == typeof(CommandBar))
+                    return parent as CommandBar;
+            }
+
+            return null;
+        }
+
+        private bool CheckAppBarButton(AppBarButton appBarButton )
+        {
+            if ( appBarButton == null)
+                return false;
+
+            if (appBarButton.IsCompact || !appBarButton.IsEnabled || appBarButton.Visibility != Visibility.Visible )
```

Comment:
Compact buttons are still visible, right? Only the label is missing.
I guess you could check for compactness before setting the textproperty in GetText().